c23068051.lua

```diff
@@ -2,61 +2,68 @@
 function c23068051.initial_effect(c)
 	--Activate
 	local e1=Effect.CreateEffect(c)
-	e1:SetDescription(aux.Stringid(23068051,0))
-	e1:SetCategory(CATEGORY_TOHAND+CATEGORY_SEARCH)
 	e1:SetType(EFFECT_TYPE_ACTIVATE)
 	e1:SetCode(EVENT_FREE_CHAIN)
 	e1:SetHintTiming(0,TIMING_END_PHASE)
+	e1:SetCost(c23068051.cost)
 	e1:SetTarget(c23068051.target)
 	e1:SetOperation(c23068051.activate)
 	c:RegisterEffect(e1)
-	--Activate2
-	local e2=Effect.CreateEffect(c)
-	e2:SetDescription(aux.Stringid(23068051,1))
-	e2:SetCategory(CATEGORY_DRAW)
-	e2:SetType(EFFECT_TYPE_ACTIVATE)
-	e2:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
-	e2:SetCode(EVENT_FREE_CHAIN)
-	e2:SetHintTiming(0,TIMING_END_PHASE)
-	e2:SetCost(c23068051.cost)
-	e2:SetTarget(c23068051.target2)
-	e2:SetOperation(c23068051.activate2)
-	c:RegisterEffect(e2)
+end
+function c23068051.cost(e,tp,eg,ep,ev,re,r,rp,chk)
+	e:SetLabel(100,0)
```

Comment:
maybe you can use [`Effect.IsCostChecked`](https://github.com/Fluorohydride/ygopro-core/pull/454)

Reply:
if `chk`'s value returned 1 in target function, `Effect.IsCostChecked` returned false.

Reply:
I don't understand, what is the problem?

Reply:
https://github.com/Fluorohydride/ygopro-core/blob/master/effect.cpp#L399
```c++
// check functions: condition, cost(chk=0), target(chk=0)
int32 effect::is_activate_ready(effect* reason_effect, uint8 playerid, const tevent& e, int32 neglect_cond, int32 neglect_cost, int32 neglect_target) {
```

so, if chk=1, Effect.IsCostChecked always return false only.

Reply:
You mean, you need to select between 2 options in the target function when the effect activate (chk=1), one of them require cost to be checked, but `Effect.IsCostChecked` always return false there, so it can't be used?

Reply:
yes.
so we'll need to update `Effect.IsCostChecked` if these scripts use `Effect.IsCostChecked`.